modules/nf-core/odgi/stats/main.nf

```diff
@@ -0,0 +1,36 @@
+process ODGI_STATS {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::odgi=0.8.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/odgi:0.8.0--py39h2add14b_0':
+        'quay.io/biocontainers/odgi:0.8.0--py310hc8f18ef_0' }"
+
+    input:
+    tuple val(meta), path(graph)
+    val(suffix)
```

Comment:
Should be passed through `$args` (see https://nf-co.re/docs/contributing/modules#general) or maybe you can obtain it from the parameters in `$args` if I understand the documentation right

Reply:
I thought the module itself should be able to run through without any optional arguments from `$args`. So the argument is mandatory in the sense that the module just won't run properly. But let me change.

Reply:
@ggabernet I discussed a lot with @mirpedrol and the usual way would be to do something like:
`def suffix = args.contains("--tsv") ? ".tsv" : ".yaml"`. However, I already need the `suffix` when describing the outputs of the module. So using `$args` is not an option. Can I still specify an additional mandatory input channel as shown in the tutorial https://nf-co.re/developers/tutorials/dsl2_modules_tutorial#create-a-test-workflow?

What other solutions would you suggest? Thanks for any help!

Reply:
As shown by @mirpedrol I am following her example https://github.com/nf-core/modules/pull/1874#pullrequestreview-1147515478 and use optional outputs. This is a straightforward solution and eliminates the requirement of an extra input channel.